lib/util/esWrapper.js

```diff
@@ -113,64 +128,114 @@ class ESWrapper {
+      }
+    }
+
+    switch (error.meta.statusCode) {
+      case 400:
+        if (kuzzleError) {
+          break;
+        }
 
-          debug('unhandled "BadRequest" elasticsearch error: %a', error);
+        if (error.meta.body && error.meta.body.error) {
+          message = error.meta.body.error.root_cause
+            ? error.meta.body.error.root_cause[0].reason
+            : error.meta.body.error.reason;
         }
 
-        kuzzleError = new BadRequestError(error);
+        kuzzleError = errorsManager.getError(
+          'external',
+          'elasticsearch',
+          'unexpected_bad_request_error',
+          message);
+
+        debug('unhandled "BadRequest" elasticsearch error: %a', error);
+
         break;
-      case 'NotFound':
-        if (!messageReplaced) {
-          if (error.body && error.body.error) {
-            message = error.body.error
-              ? error.body.error.reason + ': ' + error.body.error['resource.id']
-              : error.message + ': ' + error.body._id;
+
+      case 404:
+        if (kuzzleError) {
+          break;
+        }
+
+        // Check if it's a 404 on a document
+        if (error.meta.body.found === false) {
+          kuzzleError = errorsManager.getError(
+            'external',
+            'elasticsearch',
+            'document_not_found',
+            error.meta.body._id);
+        } else {
+          if (error.meta.body && error.meta.body.error) {
+            message = error.meta.body.error
+              ? `${error.meta.body.error.reason}: ${error.meta.body.error['resource.id']}`
+              : `${error.message}: ${error.meta.body._id}`;
           }
 
+          kuzzleError = errorsManager.getError(
+            'external',
+            'elasticsearch',
+            'unexpected_not_found_error',
+            message);
+
           debug('unhandled "NotFound" elasticsearch error: %a', error);
         }
 
-        kuzzleError = new NotFoundError(error);
         break;
-      case 'Conflict':
-        if (!messageReplaced) {
-          debug('unhandled "Conflict" elasticsearch error: %a', error);
+
+      case 409:
+        if (kuzzleError) {
+          break;
         }
 
-        kuzzleError = new ExternalServiceError(error);
+        kuzzleError = errorsManager.getError(
+          'external',
+          'elasticsearch',
+          'unexpected_conflict_error',
+          message);
+
+        debug('unhandled "Conflict" elasticsearch error: %a', error);
+
         break;
+
       default:
         debug(
           'unhandled elasticsearch error (unhandled type: %s): %a',
-          error.displayName,
+          error.meta.statusCode,
+          message);
+
+        kuzzleError = errorsManager.getError(
+          'external',
+          'elasticsearch',
+          'unexpected_error',
           message);
 
-        kuzzleError = new ExternalServiceError(error);
         break;
     }
 
-    kuzzleError.message = message;
+    // kuzzleError.stack = error.stack;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ACommentedCode)